lib/maths/CBoostedTreeImpl.cc

```diff
@@ -1445,5 +1446,54 @@ const double CBoostedTreeImpl::INF{std::numeric_limits<double>::max()};
 const CBoostedTreeHyperparameters& CBoostedTreeImpl::bestHyperparameters() const {
     return m_BestHyperparameters;
 }
+
+void CBoostedTreeImpl::computeShapValues(int topShapValues,
+                                         core::CDataFrame& frame,
+                                         const TProgressCallback&) {
+    if (m_BestForestTestLoss == INF) {
+        HANDLE_FATAL(<< "Internal error: no model available for prediction. "
+                     << "Please report this problem.");
+        return;
+    }
+    bool successful;
+    auto treeFeatureImportance =
+        std::make_unique<maths::CTreeShapFeatureImportance>(m_BestForest);
+    TDoubleVecVec shapValues;
+    TDoubleVec shapTotal;
+    size_t numberInputFields = predictionColumn(frame.numberColumns()) - 1;
+    topShapValues = (topShapValues < numberInputFields) ? topShapValues : numberInputFields;
+    std::tie(shapValues, shapTotal) =
+        treeFeatureImportance->shap(frame, *m_Encoder, numberInputFields);
+
+    // get indices of the top elements
+    std::vector<std::size_t> indices(shapTotal.size());
```

Comment:
I think we have the typedef
```suggestion
    TSizeVec indices(shapTotal.size());
```

Reply:
Fixed